R/calculate_geom_positions.R

```diff
@@ -155,6 +154,24 @@ calculate_geom_positions <- function(
       ungroup()
   })
 
+  # gather image data
+  img_data <- geom_data_processor("image", function(dat) {
+    # If the location of the image is not provided using the "value"
+    # column in the data, construct it from the directory and filename
+    print(dat$extension)
+    if (dat %has_name% "directory") {
+      dat$value <- paste0(dat$directory, dat$value, ".", dat$extension)
+    } 
```

Comment:
Print should probably be removed. Also, it might be better to process the directory and the extension separately, and to support having NA values in the column (for some reason).

```suggestion
    if (dat %has_name% "directory") {
      dat$value <- ifelse(is.na(dat$value) | is.na(dat$directory), dat$value, paste0(dat$directory, "/", dat$value))
    }
    if (dat %has_name% "extension") {
      dat$value <- ifelse(is.na(dat$value) | is.na(dat$extension), dat$value, paste0(dat$value, ".", dat$extension))
    }
```